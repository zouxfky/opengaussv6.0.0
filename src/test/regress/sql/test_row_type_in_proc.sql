create table type_test(a int, b char(10));
create table test_row_type(a int, b char(10));

create or replace procedure test_row_type_in_pro()
as declare
rec type_test%rowtype;
a int;
begin
select x.* , x2.a into rec, a from test_row_type();
end;
/

create or replace procedure test_row_type_in_pro()
as declare
rec type_test%rowtype;
a int;
begin
select x.* , x2.a into rec.*, a from test_row_type();
end;
/

create or replace procedure test_row_type_in_pro()
as declare
rec type_test%rowtype;
a int;
begin
select x.* into rec.* from test_row_type();
end;
/

create or replace procedure test_row_type_in_pro()
as declare
rec type_test%rowtype;
a int;
begin
select x.* into rec from test_row_type();
end;
/
