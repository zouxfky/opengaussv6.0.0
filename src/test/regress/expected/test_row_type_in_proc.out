create table type_test(a int, b char(10));
create table test_row_type(a int, b char(10));
create or replace procedure test_row_type_in_pro()
as declare
rec type_test%rowtype;
a int;
begin
select x.* , x2.a into rec, a from test_row_type();
end;
/
ERROR:  record or row variable cannot be part of multiple-item INTO list
LINE 5: select x.* , x2.a into rec, a from test_row_type();
                                  ^
QUERY:  declare
rec type_test%rowtype;
a int;
begin
select x.* , x2.a into rec, a from test_row_type();
end
create or replace procedure test_row_type_in_pro()
as declare
rec type_test%rowtype;
a int;
begin
select x.* , x2.a into rec.*, a from test_row_type();
end;
/
ERROR:  Improper use of '.*'. The '.*' operator cannot be used with a row type variable.
LINE 5: select x.* , x2.a into rec.*, a from test_row_type();
                                  ^
QUERY:  declare
rec type_test%rowtype;
a int;
begin
select x.* , x2.a into rec.*, a from test_row_type();
end
create or replace procedure test_row_type_in_pro()
as declare
rec type_test%rowtype;
a int;
begin
select x.* into rec.* from test_row_type();
end;
/
ERROR:  Improper use of '.*'. The '.*' operator cannot be used with a row type variable.
LINE 5: select x.* into rec.* from test_row_type();
                           ^
QUERY:  declare
rec type_test%rowtype;
a int;
begin
select x.* into rec.* from test_row_type();
end
create or replace procedure test_row_type_in_pro()
as declare
rec type_test%rowtype;
a int;
begin
select x.* into rec from test_row_type();
end;
/
